Makefile.common

```diff
@@ -304,16 +304,21 @@ prekoji-checks:
 koji: prekoji-checks kojidef
 	@$(MAKE) spdxcheck
 	@$(MAKE) checkblacklist
-	@if ! git branch | grep -q -E '^\* master'; then \
+	@if [ "$$(git rev-parse --abbrev-ref HEAD)" != "master" ]; then \
 		echo "Error: Must be on the master branch to submit to koji" >&2; \
 		exit 1; \
 	fi; \
-	if ! git diff --quiet HEAD ${SPECFILE}; then \
+	if ! git diff --quiet HEAD -- ${SPECFILE}; then \
 		echo "Error: All changes to ${SPECFILE} must be committed first" >&2; \
 		exit 1; \
 	fi; \
-	if git rev-parse --verify --quiet origin/master > /dev/null; then \
+	git fetch; \
+	if git rev-parse --verify --quiet origin/master >/dev/null; then \
 		git pull --rebase; \
+	fi; \
+	if git ls-remote --tags --exit-code origin $(SRPMVERS) >/dev/null; then \
+		echo "Error: remote tag $(SRPMVERS) already exists" >&2; \
+		exit 1; \
```

Comment:
I think I see this issue when I am doing something like killing my own push because I forgot something. If it is somebody else's release I get a conflict during the rebase before the tag check since tag is version + release and the specfile will conflict. In cases where the rebase is fine and the tag exists I would expect that the change is already pushed to git and so the next two steps of git tag and git push can be skipped and the koji build can be run (and that command will error appropriately if it is already built).

What conditions are you hitting this problem where the git tag -> git push failure is better checked here instead?

Reply:
This would happen if someone else tagged and koji'ed a version and you did not fetch/pull the remote. This fetches the freshest and checks if the tag that you intend to add has not been added. It then does not tag locally and tries to push (this latter where it would fail normally today).